core/domain/blog_domain.py

```diff
@@ -97,8 +148,12 @@ def validate(self, strict=False):
         """
         self.require_valid_title(self.title, strict)
         self.require_valid_tags(self.tags, strict)
+        if strict:
+            assert isinstance(self.thumbnail_filename, str), (
+                'Expected thumbnail filename to be a string, received %s' %
+                self.thumbnail_filename)
         self.require_valid_thumbnail_filename(
-            self.thumbnail_filename, strict=strict)
+            self.thumbnail_filename, strict=strict) # type: ignore[arg-type]
```

Comment:
Why is the ignore added?

Reply:
self.thumbnail_filename is Optional[str] but 1st argument of require_valid_thumbnail_filename is supposed to be str.

Reply:
Removed.